database_wrapper.py

```diff
@@ -349,9 +351,42 @@ def _create_database_session(self):
         self.session = Session()
         print("initialized database session")
 
+    def get_all_qa_pairs(self):
+        result = []
+
+        qa_entity = QuestionAnswerPair
+        query_session = self.session.query(qa_entity)
+        for instance in query_session:
+            result += (instance.question_format,
```

Comment:
Consider 
```
query_session = self.session.query(qa_entity.question_format, qa_entity.answer_format)
query_session.all()
``` 

because it may be a bit more efficient than a python for-loop since it just runs SQL directly and transfers less data to begin with.

Also, it would be less code and it would already return to you the list of tuple you want. 

would output
```python
[('What is the meaning of life?', 'Dr. Fizzbuzz says the answer is sqrt(1764)'), ('What is [PROFESSOR] email?', '[PROFESSOR] can be contacted at [PROFESSOR..email]')]
```

Alternative code:
```python
>>> self.session.query(qa_entity.question_format).add_column(qa_entity.answer_format).all()
[('What is the meaning of life?', 'Dr. Fizzbuzz says the answer is sqrt(1764)'), ('What is [PROFESSOR] email?', '[PROFESSOR] can be contacted at [PROFESSOR..email]')]`
```
**Resources:**
https://docs.sqlalchemy.org/en/14/orm/tutorial.html#querying
https://docs.sqlalchemy.org/en/13/orm/query.html#sqlalchemy.orm.query.Query
https://docs.sqlalchemy.org/en/13/orm/query.html#sqlalchemy.orm.query.Query.add_column

Reply:
Thanks for the suggestion, your approach is much simpler. For some reason I couldn't get the properties after querying the entity and had to resort the looping technique suggested in a StackOverflow discussion